OpenRA.Mods.Common/DiscordService.cs

```diff
@@ -0,0 +1,101 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using DiscordRPC;
+using DiscordRPC.Logging;
+
+namespace OpenRA.Mods.Common
+{
+	public class DiscordService : IGlobalModData
+	{
+		public readonly string ApplicationId = null;
+		public readonly string Name = null;
+		public readonly string LargeImageKey = null;
+
+		readonly DiscordRpcClient client;
+		RichPresence richPresence;
+
+		static readonly Lazy<DiscordService> Service = Exts.Lazy(() =>
+		{
+			if (!Game.ModData.Manifest.Contains<DiscordService>())
+				return null;
+
+			var discordService = Game.ModData.Manifest.Get<DiscordService>();
+
+			return discordService;
+		});
+
+		public DiscordService(MiniYaml yaml)
+		{
+			FieldLoader.Load(this, yaml);
+
+			client = new DiscordRpcClient(ApplicationId, autoEvents: true);
+
+			client.Initialize();
+		}
+
+		void SetStatus(DiscordState state)
+		{
+			var party = state.Players.HasValue && state.Slots.HasValue ? new Party
```

Comment:
Style nit: we generally try to avoid hanging braces to avoid ambiguity. In other cases like this we have typically done:
```csharp
var party = state.Players.HasValue && state.Slots.HasValue ? new Party
{
	ID = Secrets.CreateFriendlySecret(new Random()),
	Size = state.Players.Value,
	Max = state.Slots.Value
} : null;
```

much like `richPresence` just below.

Reply:
Severity	Code	Description	Project	File	Line	Suppression State
Error	SA1500	Braces for multi-line statements should not share line	OpenRA.Mods.Common	C:\projects\OpenRA\OpenRA.Mods.Common\DiscordService.cs	51	Active